app/mailers/user_notifications.rb

```diff
@@ -10,11 +10,15 @@ class UserNotifications < ActionMailer::Base
 
   include Email::BuildEmailHelper
 
-  def signup(user, opts={})
-    build_email(user.email,
-                template: "user_notifications.signup",
-                locale: user_locale(user),
-                email_token: opts[:email_token])
+  %i(signup
+     confirm_old_email
+     confirm_new_email
+     email_login
+     admin_login
+     account_created).each do |mail_name|
+    define_method(mail_name) do |user, opts={}|
+      build_email_with_email_token(user, "user_notifications.#{mail_name}", opts[:email_token])
```

Comment:
I'll prefer if we avoid using meta-programming techniques here. Extracting the common logic into a private method is a good enough refactor.